taraxa/state/dpos_2.0/precompiled/iterable/iterable_map.go

```diff
@@ -0,0 +1,112 @@
+package iterable
+
+import (
+	"math/big"
+
+	"github.com/Taraxa-project/taraxa-evm/common"
+	"github.com/Taraxa-project/taraxa-evm/taraxa/util/keccak256"
+	"github.com/Taraxa-project/taraxa-evm/taraxa/util/storage"
+)
+
+type StorageData = map[*common.Hash][]byte
+
+type Serializable interface {
+	// Serialize() []byte
+	Serialize(pos *common.Hash, out StorageData)
+}
+
+type Map[T Serializable] struct {
+	storage          storage.StorageWrapper
+	storage_position *common.Hash
+	keys             []common.Address
+	mapping          map[common.Address]T
```

Comment:
This is not ideal because you have to iterate over keys to find specific address when deleting (`findIndex`), in solidity code there is never need to iterate over array to find if the key is in it. I think the solution here is pretty easy, we could add another mapping here:
```
keys                  []common.Address
keys_mapping  map[common.Address]uint64 // mapping with indexes for keys array
mapping          map[common.Address]T          // mapping with real data
```

With this structure:

- no need to iterate over keys because you can find out instantly if specific address is in keys by looking into the keys_mapping 
- on all places where we wanna check there is an item for specidiv address in iterbale map, we should check keys_mapping, not mapping
- mapping is used just to get the final data, no checks or anything else
- when deleting item, we have to delete it in keys & keys_mapping as well as mapping